qt/aqt/operations/scheduling.py

```diff
@@ -65,10 +66,37 @@ def set_due_date_dialog(
 
 
 def forget_cards(
-    *, parent: QWidget, card_ids: Sequence[CardId]
-) -> CollectionOp[OpChanges]:
+    *,
+    parent: QWidget,
+    card_ids: Sequence[CardId],
+    context: ScheduleCardsAsNew.Context.V | None = None,
+) -> CollectionOp[OpChanges] | None:
+    assert aqt.mw
+
+    dialog = QDialog(parent)
+    disable_help_button(dialog)
+    form = aqt.forms.forget.Ui_Dialog()
+    form.setupUi(dialog)
+
+    if context is not None:
+        defaults = aqt.mw.col.sched.schedule_cards_as_new_defaults(context)
+        form.restore_position.setChecked(defaults.restore_position)
+        form.reset_counts.setChecked(defaults.reset_counts)
+
+    if not dialog.exec():
```

Comment:
Ideally we should avoid .exec() calls, instead calling .open() after attaching to the `accepted` slot. It's a bit more cumbersome, but avoids some issues:

https://github.com/ankitects/anki/pull/1684#discussion_r813434377

Ideally we should also be querying the defaults with QueryOp(), but unfortunately that doesn't play well with our current approach of returning a CollectionOp so that the caller can adjust the success callback.